std/format/internal/write.d

```diff
@@ -3011,7 +3011,7 @@ void enforceValidFormatSpec(T, Char)(scope const ref FormatSpec!Char f)
 /*
     `enum`s are formatted like their base value
  */
-void formatValueImpl(Writer, T, Char)(auto ref Writer w, const(T) val, scope const ref FormatSpec!Char f)
```

Comment:
No. Again, this bug has nothing to do with `toString`; please read [the bugzilla report][1] and/or the unit test carefully so that you understand what is actually causing the issue here.

[1]: https://issues.dlang.org/show_bug.cgi?id=23400

Reply:
Ok, I see now. 

Would qualifying `val` as `auto ref` alleviate the problem?

Reply:
Ahh, no.

Reply:
The call to 

```d
foreach (i, e; EnumMembers!T)
```

looks like a code smell to me.

Can anybody explain why it is needed?

Reply:
I can't find one.

Reply:
I found a way to avoid the call to `EnumMembers` at https://github.com/dlang/phobos/pull/8593.

Reply:
Why would using `EnumMembers` be a problem here?

Reply:
I agree with @pbackus I don't see why this is a problem. This is an unrolled loop, with the compiler able to optimize as much as possible. Changing to something else might speed up compilation at the expense of runtime performance.

Reply:
The motive is to avoid template bloat.

Reply:
As I mentioned in the discussion of #8593, `EnumMembers` only incurs a single template instantiation now that it has been rewritten to use alias assignment instead of recursion. Do you have any data to support the claim that this single instantiation results in an unacceptable amount of template bloat?

Reply:
AliasSeq is still a template.